ophyd/areadetector/filestore_mixins.py

```diff
@@ -241,25 +247,66 @@ def write_path_template(self, val):
         self._write_path_template = _ensure_trailing_slash(val)
 
     def stage(self):
-        self._point_counter = count()
+        self._datum_counter = count()
         self._locked_key_list = False
         self._datum_uids.clear()
         super().stage()
         if self.enable.get() and self._reg is None:
             raise RuntimeError('The plugin {!r} is enabled, but '
                                'the Registry (self._reg) is `None`')
+        # Subclasses will assemble resource_kwargs and call
+        # self._generate_resource(resource_kwrags)
+
+    def _generate_resource(self, resource_kwargs):
```

Comment:
do we want to move this method and all it's state to a stand-alone class in databroker?

Reply:
I agree and would say the same for datum, if I understand you @tacaswell .
In my opinion, this would help a newcomer better understand the document structure (by inspecting the generation functions). It won't look nice, but might help when we have libraries other than ophyd which may generate documents.

Reply:
ah, good point about not wanting to have a databroker dependency in ophyd.  Should probably be factored out into a helper class though.

Reply:
actually i was saying the opposite, it should use a databroker function since it is databroker, but you're right, that adds a dependency. in the end, that's one reason there are schemas for these i guess. i think we've just both self-countered our arguments.